osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,184 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <AccCtrl.h>
+#include <Aclapi.h>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <windows.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem.hpp>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace alg = boost::algorithm;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// map to get access mode string
+static std::unordered_map<ACCESS_MODE, std::string> accessModeToStr = {
+    {NOT_USED_ACCESS, "Not Used"},
+    {GRANT_ACCESS, "Grant"},
+    {SET_ACCESS, "Set"},
+    {DENY_ACCESS, "Deny"},
+    {REVOKE_ACCESS, "Revoke"},
+    {SET_AUDIT_SUCCESS, "Set Audit Success"},
+    {SET_AUDIT_FAILURE, "Set Audit Failure"}};
+
+// map to build access string
+static std::map<unsigned long, std::string> permVals = {
+    {DELETE, "Delete"},
+    {READ_CONTROL, "Read Control"},
+    {WRITE_DAC, "Write DAC"},
+    {WRITE_OWNER, "Write Owner"},
+    {SYNCHRONIZE, "Synchronize"},
+    {STANDARD_RIGHTS_REQUIRED, "Std Rights Required"},
+    {STANDARD_RIGHTS_ALL, "Std Rights All"},
+    {SPECIFIC_RIGHTS_ALL, "Specific Rights All"},
+    {ACCESS_SYSTEM_SECURITY, "Access System Security"},
+    {MAXIMUM_ALLOWED, "Maximum Allowed"},
+    {GENERIC_READ, "Generic Read"},
+    {GENERIC_WRITE, "Generic Write"},
+    {GENERIC_EXECUTE, "Generic Execute"},
+    {GENERIC_ALL, "Generic All"}};
+
+// map to get inheritance string
+static std::unordered_map<unsigned long, std::string> inheritanceToStr = {
+    {CONTAINER_INHERIT_ACE, "Container Inherit Ace"},
+    {INHERIT_NO_PROPAGATE, "Inherit No Propagate"},
+    {INHERIT_ONLY, "Inherit Only"},
+    {OBJECT_INHERIT_ACE, "Object Inherit Ace"},
+    {SUB_CONTAINERS_AND_OBJECTS_INHERIT, "Sub containers and Objects Inherit"},
+    {NO_INHERITANCE, "No Inheritence"}};
+
+// helper function to build access string from permission bit mask
+std::string accessPermsToStr(const unsigned long pmask) {
+  std::vector<std::string> permList;
+
+  for (auto const& perm : permVals) {
+    if ((pmask & perm.first) != 0) {
+      permList.push_back(perm.second);
+    }
+  }
+
+  return alg::join(permList, ",");
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& trustee) {
+  // Max username length for Windows
+  int r;
```

Comment:
You don't really need to define `r` outsize of using scope. You can use smth like:
```c++
auto r = LookupAccountSid(...);
if (!r) { ...
```